CONTRIBUTING.md

```diff
@@ -36,61 +34,210 @@ $ sudo dnf install device-mapper-devel btrfs-progs-devel
+image with the broker, there are a few things worth pointing out:
+
+- The `broker` image is copied to `/usr/bin/asbd`.
+- The home directory in the image is `/opt/ansibleservicebroker`.
+- The directory `/etc/ansible-service-broker` is where, by default, the broker's configuration
+  file is mounted when deployed in OpenShift. Looking at the
+  [deploy-ansible-service-broker template](templates/deploy-ansible-service-broker.template.yaml)
+  you can find the `broker-config` included  as `config-volume` in the `asb` container and mounted at
+  `/etc/ansible-service-broker`.
+- The entrypoint to the image is [entrypoint.sh](build/entrypoint.sh). This script is what starts
+  the broker when deployed as a container.
+
+Once you have a built Docker image, you can [deploy broker from image](#deploy-broker-from-image)
+with `make deploy` **BUT** only after you have pushed your image to the registry
+(ie. `docker push ${REGISTRY}/${ORG}/origin-ansible-service-broker:${TAG}`).
+
+
+# Run Your Broker
+
+Once you have [built the Ansible Service Broker](#build-the-ansible-service-broker), there
+are a couple of options for running, testing, and debugging the broker. You can
+[run the broker binary locally](#run-your-broker-locally) **or**, assuming you
+[built a Docker image](#package-your-broker-using-docker), you can
+[deploy your broker](#deploy-broker-from-image) into a running cluster.
+
+## Run Your Broker Locally
+
+If you are interested in rapidly iterating over your broker changes or want to easily keep
+tabs on what the broker is doing, running the broker locally is an excellent idea.
+
+Before attempting to run the broker locally, you will need to have deployed an
+[OpenShift Origin Cluster with the Ansible Service Broker](#deploy_an_openshift_origin_cluster_with_the_ansible_service_broker).
+You can verify this by running `oc get all --all-namespaces` and looking for the `ansible-service-broker`
+`service-catalog` running in the cluster.
+
+**Configuration**
+
+Next, you will need configure your local development variables:
+
+```
+cp scripts/my_local_dev_vars.example scripts/my_local_dev_vars
+```
+
+Now you can modify `scripts/my_local_dev_vars` with things like your `DOCKERHUB_USERNAME`
+or use an insecure broker with `BROKER_INSECURE="true"`.
 
-**Config**
-A broker is configured via the `config.yaml` file. It's recommended to
-copy over `etc/example-config.yaml` to `etc/ansible-service-broker/config.yaml`, and edit
-as desired.
+**Prepare Local Environment**
 
-See the [Broker Configuration](docs/config.md) doc for other example
-configurations.
+Running `make prepare-local-env` will do several things on your behalf:
+
+* Remove the running `ansible-service-broker` from the cluster, leaving only
+  etcd running in the namespace.
+* Modify the `asb` service to point where our locally running broker will be.
+* Generate a configuration file for the broker (`etc/generated_local_development.yaml`)
```

Comment:
CHANGE: broker (`etc/generated_local_development.yaml`) -> broker, `etc/generated_local_development.yaml`. Basically remove the parens and add a comma.